storage/storage.cpp

```diff
@@ -1319,6 +1319,13 @@ void Storage::GetNodeAttrs(TCountryId const & countryId, NodeAttrs & nodeAttrs)
       countryIdAndName.m_localName = m_countryNameGetter(countryIdAndName.m_id);
     nodeAttrs.m_parentInfo.emplace_back(move(countryIdAndName));
   }
+  // Parents country.
+  nodeAttrs.m_topmostParentInfo.clear();
+  ForEachAncestorExceptForTheRoot(nodes, [&](TCountryId const & ancestorId, TCountryTreeNode const & node)
+  {
+    if (node.Value().GetParent() == GetRootId())
+      nodeAttrs.m_topmostParentInfo.push_back({ancestorId, m_countryNameGetter(ancestorId)});
```

Comment:
Тут явно место для emplace_back, тогда будут не нужны { }

> On Mar 28, 2016, at 13:27, Vladimir Byko-Ianko <notifications@github.com> wrote:
> 
> In storage/storage.cpp <https://github.com/mapsme/omim/pull/2554#discussion_r57560521>:
> 
> > @@ -1319,6 +1319,13 @@ void Storage::GetNodeAttrs(TCountryId const & countryId, NodeAttrs & nodeAttrs)
> >        countryIdAndName.m_localName = m_countryNameGetter(countryIdAndName.m_id);
> >      nodeAttrs.m_parentInfo.emplace_back(move(countryIdAndName));
> >    }
> > +  // Parents country.
> > +  nodeAttrs.m_topmostParentInfo.clear();
> > +  ForEachAncestorExceptForTheRoot(nodes, [&](TCountryId const & ancestorId, TCountryTreeNode const & node)
> > +  {
> > +    if (node.Value().GetParent() == GetRootId())
> > +      nodeAttrs.m_topmostParentInfo.push_back({ancestorId, m_countryNameGetter(ancestorId)});
> Если делать, как ты предлагаешь, придется писать два конструктора у CountryIdAndName дефолтный и от 2 параметров. Получится достаточно многословно. А если их писать, то тогда уж не явно указывать тип, а лучше, воспользоваться методом emplace_back(ancestorId, m_countryNameGetter(ancestorId)).
> 
> С другой стороны, в С++11 введено много вещей, которые не позволяют указать явно тип при создании объекта. Мы используем списки инициализации, auto методы emplace_back и т.п.
> 
> Не смотря на минус, указанный тобой, я все же за список инициализации здесь.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2554/files/7de6c3345031add7c5f09e4efaf874f0ffdd8a5f#r57560521>